### PR TITLE
Update Elixir on GAE tutorial to cover Phoenix 1.4 and Cloud SQL

### LINKED ISSUE #358: Tutorial Request: Run an Elixir Phoenix app WITH ECTO on the Google App Engine Flexible Environment
There is no resources to find how to get phoenix app with ecto running (postgres)  

Technical Level
intermediate

Length
short (< 250 words)

### PR DESCRIPTION
**Pull request by dazuma**
This is a long-needed update to the Elixir Phoenix on GAE tutorial.

It updates the tutorial to cover recent updates to the Phoenix framework (version 1.4) and the Distillery build tool (version 2.0). More significantly, it now includes material on how to connect to a Postgres database running in Cloud SQL, a very common request from readers.

Fixes #358

### CODE REVIEW

**Comment by dazuma** [comment]
Putting a brief hold on this. I'm rearranging the database section for clarity.

**Comment by ToddKopriva** [comment]
Thanks, @dazuma . 

@steveperry-53 , feel free to pass the editorial review to me if you like.

**Comment by dazuma** [comment]
Note: I changed the local cloudsql directory path from `tmp/cloudsql` (relative to the user's application directory) to `/tmp/cloudsql` (in the root tmp directory). This is because psql wants an absolute path in order to recognize it as a unix socket. (Found while testing the similar change in #622)

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 1]
```diff
@@ -16,12 +16,15 @@ Engine in minutes.
 
 This tutorial will help you get started deploying a Phoenix app to App Engine.
 You will create a new Phoenix application, and learn how to configure, deploy,
-and update it. For simplicity, this tutorial app will not use Ecto or connect
-to a SQL database, but you can extend it to connect to Google Cloud SQL or any
-other database service.
+and update it. The application will also demonstrate how to connect to a
+PostgreSQL database running on Cloud SQL.
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Suggestion: On first mention, link Cloud SQL to https://cloud.google.com/sql.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 2]
```diff
@@ -32,11 +35,16 @@ Before running this tutorial, take the following steps:
...
-3.  Install the [Google Cloud SDK](https://cloud.google.com/sdk/). Make sure
+3.  [Enable the Google Cloud SQL Admin API in the Cloud Console](https://console.cloud.google.com/apis/library/sqladmin.googleapis.com/).
+
+4.  Install the [Google Cloud SDK](https://cloud.google.com/sdk/). Make sure
     you [initialize](https://cloud.google.com/sdk/docs/initializing) the SDK
     and set the default project to the new project you created.
 
-    Version 175.0.0 or later of the SDK is required.
+    Version 227.0.0 or later of the SDK is required. If you have an earlier
+    version installed, you may upgrade it by running
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Suggestion: Make punctuation at the end of lead-in text consistent. This topic sometimes has no punctuation, sometimes a period, and sometimes a colon. I'd suggest using a colon.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 3]
```diff
@@ -48,37 +56,238 @@ If you have not yet installed Elixir and Phoenix, do so:
...
+6.  Visit [http://localhost:4000](http://localhost:4000) to verify that your
+    new code is running. You can log into your database and add new rows, and
+    reload the page to verify that the count has changed.
+
+For more information on using Ecto to access a SQL database, see the
+[Phoenix Ecto guide](https://hexdocs.pm/phoenix/ecto.html).
+
+## Create a production database in Cloud SQL
+
+In this section, you will create your "production" database using Cloud SQL, a
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: No need for quotes around "production".

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 4]
```diff
@@ -48,37 +56,238 @@ If you have not yet installed Elixir and Phoenix, do so:
...
+
+### Create a Cloud SQL instance
+
+First you will create a new database in the cloud.
+
+1.  Create a Cloud SQL instance named `hellodb` with a Postgres database
+    by running the following command:
+
+        gcloud sql instances create hellodb --region=us-central \
+          --database-version=POSTGRES_9_6 --tier=db-g1-small
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: I don't think us-central is a region. We do have a region named us-central1. Occurs more than once in this topic.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 5]
```diff
@@ -48,37 +56,238 @@ If you have not yet installed Elixir and Phoenix, do so:
...
+
+    You may choose a region other than `us-central` if there is one closer to
+    your location.
+
+2.  Get the _connection name_ for your Cloud SQL instance by running the
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Consider reducing the number of italicized terms. For example, I don't think the italics are needed here for connection name.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 6]
```diff
@@ -48,37 +56,238 @@ If you have not yet installed Elixir and Phoenix, do so:
...
+is useful for the initial creation and migration of your database, as well as
+for creating _ad hoc_ database connections for maintenance.
+
+By default, Cloud SQL instances are secured: to connect using the standard
+`psql` tool, you must whitelist your IP address. This security measure can make
+it challenging to establish _ad hoc_ database connections. So, Cloud SQL
+provides a command line tool called the
+[Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy). This
+tool communicates with your database instance over a secure API, using your
+Cloud SDK credentials, and opens a local endpoint (such as a unix socket) that
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Capitalize Unix. Occurs a few times.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 7]
```diff
@@ -153,26 +335,32 @@ Now you will deploy your new app to App Engine.
...
+    needed to connect to the database.
 
 2.  Run the following command to deploy your app:
 
         gcloud app deploy
 
     If this is the first time you have deployed to App Engine in this project,
-    gcloud will prompt you for a region. You may choose the default of
-    "us-central", or select a region close to your location.
+    gcloud will prompt you for a region. It is generally a good idea to choose
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Code font for `gcloud`. Backticks.

File: tutorials/elixir-phoenix-on-google-app-engine.md [hunk 8]
```diff
@@ -48,37 +56,238 @@ If you have not yet installed Elixir and Phoenix, do so:
...
+In this section, you will create a new Phoenix app with a database, and make
+sure it runs locally in development. If you already have an app to deploy, you
+may use it instead.
 
-1.  Run the `phx.new` task to create a new Phoenix project called
-    "appengine_example". This tutorial will omit Ecto for now.
+### Create a new Phoenix app
 
-        mix phx.new appengine_example --no-ecto
+1.  Run the `phx.new` task to create a new Phoenix project called "hello".
```
Comment (steveperry-53) [inline tutorials/elixir-phoenix-on-google-app-engine.md]: Elsewhere in the topic, you use code font for the name of the app: `hello`. Suggest doing the same  here.

**Comment by ToddKopriva** [comment]
I'm going to do a quick pass to make the Markdown comply with our publishing system, and then I'll squash the commits and merge the PR. I expect to republish the /community site with this change today.

**Comment by dazuma** [comment]
Removed the angle brackets since they aren't strictly necessary.

**Comment by ToddKopriva** [comment]
Thanks, Daniel. I'll do the production pass today.

**Pull request merged**